src/v/cloud_storage/remote.h

```diff
@@ -354,10 +357,15 @@ class remote : public ss::peering_sharded_service<remote> {
     /// to perform sequential deletes of the objects.
     ///
     /// \param bucket The bucket to delete from
-    /// \param keys A vector of keys which will be deleted
+    /// \param keys A range of keys which will be deleted
+    template<typename R>
+    requires std::ranges::range<R>
+             && std::same_as<
+               std::ranges::range_value_t<R>,
+               cloud_storage_clients::object_key>
     ss::future<upload_result> delete_objects(
       const cloud_storage_clients::bucket_name& bucket,
-      std::vector<cloud_storage_clients::object_key> keys,
+      R keys,
```

Comment:
little confused about why we are templating on a range type here. aren't we always passing in a single container type?